### PR TITLE
Add new functional

### PR DESCRIPTION
**Pull request by sapozhnykov**

### CODE REVIEW

File: src/main/java/com/sapozhnykov/App.java [hunk 1]
```diff
@@ -1,13 +1,49 @@
...
+
+        MenuView catalogMenu = new CatalogMenuImpl(productService, orderService);
+        MenuView clientListMenu = new ClientListMenuImpl(clientService, authClientService);
+        MenuView productListMenu = new ProductListMenuImpl(productService);
+        MenuView orderListMenu = new OrderListMenuImpl(orderService);
+        MenuView clientMenu = new ClientMenuImpl(catalogMenu, clientService);
+        MenuView authClientMenu = new AuthClientMenuImpl(clientMenu, authClientService, clientService);
+        MenuView adminMenu = new AdminMenuImpl(clientListMenu, productListMenu, orderListMenu);
+
+        MenuView mainMenu = new MainMenuImpl(authClientMenu, adminMenu);
```
Comment (aleksashyn) [inline src/main/java/com/sapozhnykov/App.java]: Try to apply service locator

File: src/main/java/com/sapozhnykov/dao/auth/Impl/AuthClientDaoImpl.java [hunk 2]
```diff
@@ -7,10 +7,12 @@
 import java.util.List;
 
 public class AuthClientDaoImpl implements AuthClientDao {
+    private static AuthClientDao authClientDao = new AuthClientDaoImpl();
```
Comment (aleksashyn) [inline src/main/java/com/sapozhnykov/dao/auth/Impl/AuthClientDaoImpl.java]: It's can be final

File: src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java [hunk 3]
```diff
@@ -7,17 +7,20 @@
...
+
     private List<Client> clients = new ArrayList<>();
-    private static long tempID = 1;
+    private static long tempID = 3;
 
-    public ClientDaoImpl() {
-        clients.add(new Client(tempID++,"David" , "Cramer", "0503213232", "david@gmail.com"));
-        clients.add(new Client(tempID++,"Greg" , "Popovich", "0974321222", "greg@gmail.com"));
+    private ClientDaoImpl() {
+        clients.add(new Client(1,"David" , "Cramer", "0503213232", "david@gmail.com"));
```
Comment (aleksashyn) [inline src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java]: It's bed idea to fulfilled data inside contractor

File: src/main/java/com/sapozhnykov/dao/order/impl/OrderDaoImpl.java [hunk 4]
```diff
@@ -8,17 +8,43 @@
...
     public boolean add(long clientId, ArrayList<Product> products) {
         Order order = new Order(tempID++, clientId, products);
         return orders.add(order);
     }
 
+    @Override
+    public boolean deleteById(long id) {
+        Order tempOrder;
+
+        tempOrder = getById(id);
```
Comment (aleksashyn) [inline src/main/java/com/sapozhnykov/dao/order/impl/OrderDaoImpl.java]: Please, merge declaration and initialization

File: src/main/java/com/sapozhnykov/domain/Order.java [hunk 5]
```diff
@@ -6,7 +6,7 @@
 public class Order {
     private long id;
     private long clientId;
-    private List<Product> products;
+    private ArrayList<Product> products;
```
Comment (aleksashyn) [inline src/main/java/com/sapozhnykov/domain/Order.java]: Why you use concrete class for type?

File: src/main/java/com/sapozhnykov/App.java (hunk 1 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/App.java]: Fixed

File: src/main/java/com/sapozhnykov/dao/auth/Impl/AuthClientDaoImpl.java (hunk 2 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/dao/auth/Impl/AuthClientDaoImpl.java]: Fixed

File: src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java (hunk 3 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java]: Fixed

File: src/main/java/com/sapozhnykov/dao/order/impl/OrderDaoImpl.java (hunk 4 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/dao/order/impl/OrderDaoImpl.java]: Fixed

File: src/main/java/com/sapozhnykov/domain/Order.java (hunk 5 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/domain/Order.java]: Fixed

File: src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java (hunk 3 above)
Comment (aleksashyn) [reply src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java]: I mean that each time when you create instance you fulfilled data, it's better to  extract this from class at all

File: src/main/java/com/sapozhnykov/services/locator/impl/LocatorServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import com.sapozhnykov.services.order.OrderService;
+import com.sapozhnykov.services.order.impl.OrderServiceImpl;
+import com.sapozhnykov.services.product.ProductService;
+import com.sapozhnykov.services.product.impl.ProductServiceImpl;
+import com.sapozhnykov.validators.ValidationService;
+import com.sapozhnykov.validators.impl.ValidationServiceImpl;
+import com.sapozhnykov.view.menu.MenuView;
+import com.sapozhnykov.view.menu.impl.*;
+
+public class LocatorServiceImpl implements LocatorService {
```
Comment (aleksashyn) [inline src/main/java/com/sapozhnykov/services/locator/impl/LocatorServiceImpl.java]: where you find this example? It's not ServiceLocator

File: src/main/java/com/sapozhnykov/domain/Order.java (hunk 5 above)
Comment (aleksashyn) [reply src/main/java/com/sapozhnykov/domain/Order.java]: please, check all you code for this

File: src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java (hunk 3 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/dao/client/impl/ClientDaoImpl.java]: okay, fixed

File: src/main/java/com/sapozhnykov/domain/Order.java (hunk 5 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/domain/Order.java]: okay, fixed

File: src/main/java/com/sapozhnykov/services/locator/impl/LocatorServiceImpl.java (hunk 6 above)
Comment (sapozhnykov) [reply src/main/java/com/sapozhnykov/services/locator/impl/LocatorServiceImpl.java]: okay, fixed

**Review by aleksashyn — APPROVED** [review]

**Pull request merged**